src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs

```diff
@@ -387,9 +400,10 @@ protected void CheckFileTypeUsage(TypeWithAnnotations returnType, ImmutableArray
             bool isExtensionMethod,
             bool isNullableAnalysisEnabled,
             bool isVarArg,
-            bool isExplicitInterfaceImplementation)
+            bool isExplicitInterfaceImplementation,
+            bool hasThisInitializer = false)
```

Comment:
>bool hasThisInitializer = false) [](http://example.com/codeflow?start=12&length=32)

Optional parameters are bug prone. It is easy to fail to pass value for them. I understand that `this` initializers are not common, but I think it is better to have no optional parameters in this API.

Reply:
@AlekseyTs sgtm.